packages/overlay/stories/overlay.stories.ts

```diff
@@ -1413,3 +1413,70 @@ export const virtualElement = (args: Properties): TemplateResult => {
 virtualElement.args = {
     placement: 'right-start' as Placement,
 };
+
+export const virtualElementDeclaratively = (
+    args: Properties
+): TemplateResult => {
+    const handleContextmenu = async (event: PointerEvent): Promise<void> => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        const overlay = document.querySelector('#overlay') as Overlay;
+
+        const virtualTrigger = new VirtualTrigger(event.clientX, event.clientY);
```

Comment:
I think we want to have a stable `virtualTrigger`. This would allow us to attach it to the `<sp-overlay>` via `.trigger=${virtualTrigger}` and then leverage `updateBoundingClientRect()` on the virtual trigger to update the position, which also calls `Overlay.update()` for you under the hood.